### DISCUSSION TITLE
How to install a pytorch2.4.1 compitable whisper?

### DISCUSSION THREAD

**Discussion by JohnHerry** [openai/whisper#2350] [Q&A]
I use the official `pip install -U openai-whisper`  , it will download and install torch2.4.0 and torch2.3.1 and triton2.3.1;  My env is python3.10 and torch2.4.1 and default to triton3.0, is there any publish of whisper that will not remove my local torch and triton installation? and it is some what confuse me that why my installation will download two versions of torch?

**Comment by Et0h** [comment]
I have used Whisper with Python 3.12.6 (64-bit) on Windows 10 with Torch 2.4.1. I did something like this:
1. I downloaded the ffmpeg executables and put them in the folder where my media was that I wanted to transcribe.
2. I downloaded Python from https://www.python.org/ftp/python/3.12.6/python-3.12.6-amd64.exe - for other builds in that version see https://www.python.org/ftp/python/3.12.6/
3. I installed it for all users to `C:\Python312`
4. In command line I ran `python.exe -m venv venv-3.12`
5. In command line I went to the `venv-3.12` subdirectory of the Python directory and ran `activate.bat`
6. In the VENV I ran `pip install openai-whisper`
7. In the VENV I ran `CD c:\mediadir\`
8. In the VENV I ran `whisper --language English "filename.mp3"`

I can't speak to Triton.

**Comment by JohnHerry** [comment]
thank you for help
